controllers/apps/operations/restart.go

```diff
@@ -72,30 +70,38 @@ func (r restartOpsHandler) Action(reqCtx intctrlutil.RequestCtx, cli client.Clie
 		return err
 	}
 	r.compOpsHelper = newComponentOpsHelper(opsRes.OpsRequest.Spec.RestartList)
-	componentKindList := []client.ObjectList{
-		&appv1.StatefulSetList{},
-		&workloads.InstanceSetList{},
+	orderedComps, err := r.getComponentOrders(reqCtx, cli, opsRes)
+	if err != nil {
+		return err
 	}
-	for _, objectList := range componentKindList {
-		if err := r.restartComponent(reqCtx, cli, opsRes, objectList); err != nil {
-			return err
-		}
+	if len(orderedComps) > 0 {
+		return r.restartComponents(reqCtx, cli, opsRes, orderedComps[:1], false)
```

Comment:
why split the restart ops into two steps? restart the first component in Action and others in ReconcileAction?

Reply:
I mean restart all components in ReconcileAction seems easier to understand

Reply:
When restarting a single component, it's possible to skip the reconcile logic.

Reply:
improved